.github/workflows/test_java_osx_cookbook.yml

```diff
@@ -37,6 +37,8 @@ jobs:
       matrix:
         python-version: [ '3.9' ]
         java-version: [ '11', '17', '18' ]
+        compiler-version: ['11']
+        arrow-version: ['9.0.0']
```

Comment:
I still don't understand why we're parameterizing this in CI? It just makes it harder to update Arrow.

Reply:
The main reason is to handle that as a matrix, this support pass an arrays of arrow_versions needed on demand, I could suggest that new changes related to a version would be handled by this CI file without change main code.

Reply:
I don't think we should do this until we decide how we want to approach multiple Cookbook versions.

Reply:
At the very least, I'd rather we put these CI changes separately instead of combining them with a larger change

Reply:
Make sense, let me replace with the last CI configuration